### DISCUSSION TITLE
Calibration help

### DISCUSSION THREAD

**Discussion by talormanda** [tronikos/esphome-magnetometer-water-gas-meter#43] [Q&A] [answered]
Do you need to adjust a setting besides the Calibrate axis button? It did work when I pressed it, and the values were pretty consistently the same. I did measure out 12 cups of water which should be 0.75 gal, but my total gallons used was only changed by 0.46. This was coming from the sink which is connected to a water heater.

Do I need to adjust the volume per half rotation number?

https://muellersystems.com/wp-content/uploads/2021/04/14205-420.Composite_blackbase_Jun2020-Jen-1.pdf

![image](https://github.com/user-attachments/assets/fa1944a6-bbe0-4599-a188-9e890bb704c0)
![image](https://github.com/user-attachments/assets/222734e5-bbb1-4622-9b0a-9e400a829adb)

**Comment by tronikos** [comment]
Yes you need to set a good value for the Volume per half rotation. See https://github.com/tronikos/esphome-magnetometer-water-gas-meter?tab=readme-ov-file#volume-per-half-rotation

**Reply by talormanda** [reply]
if my meter is 5/8. is it as simple as 0.00864902 (2 / 231.24) ? Or is it better to do everything below that in the "Alternatively" part?

**Reply by tronikos** [reply, marked as answer]
It would have been 2 / 231.24 only if you had a Neptune T-10. Follow the "Alternatively" part.

**Reply by talormanda** [reply]
"4. Set this to the result of: diff of readings in volume_unit divided by diff of half rotations."

this step, just to make sure I understand. it is:
`(new_meter_reading - old_meter_reading) / (new_half_rotation_number - old_half_rotation_number)`  ?

**Reply by tronikos** [reply]
Correct.

**Reply by talormanda** [reply]
Do you happen to know how to read the meter? Does the very last 0 represent the number on the arrow below it?

**Reply by talormanda** [reply]
Well I was able to let it run for a few hours and get a good number. It did measure the same amount of water I emptied out of the sink.
